docs/tech-reference/rpm.rst

```diff
@@ -0,0 +1,452 @@
+content come from? Who is generating it, and with what tools? The answer to
+these questions are answered in this document. Note that this document
+only covers Fedora and Red Hat Enterprise Linux. It does not cover SLES,
+OpenSuse, Scientific Linux, CentOS, Oracle Linux, or any other RPM-based
+distributions (yet).
+
+
+The RPM
+-------
+
+RPM is a packaging format. Many Linux distributions use RPM to distribute their
+software packages, although more do not. An RPM is a collection of metadata
+about the package as well as a payload. What the payload contains depends on the
+type of RPM. There are three different types of RPM:
+
+ * Source code RPMs (referred to as "SRPMs", ".src.rpm" file extension)
+
+ * Binary RPMs (referred to as "RPMs", ".rpm" file extension)
+
+ * Binary delta RPMs (referred to as "DRPMs", ".drpm" file extension)
+
+
+File Format
+^^^^^^^^^^^
+
+The RPM file format consists of a "lead" which identifies the file as an RPM, a
+"signature" section which can be used to ensure the integrity and authenticity
+of the RPM (via GPG), a "header" section which contains metadata about the
+packaged software (name, version, architecture, file list, etc.), and a
+"payload" which is a file archive (usually in the cpio format) compressed with
+gzip.
+
+SRPM
+^^^^
+
+The payload of a source RPM is simply a compressed tarball of source code and a
+file, called a spec file, that describes how to turn the source RPM into a
+binary RPM. The source RPM can be built into one or more binary RPMs using
+``rpmbuild``.
+
+RPM
+^^^
+
+A binary RPM's payload is the collection of files installed from the build
+process of the source code. The spec file in the source RPM is responsible for
+the locations and permissions of each file. This allows the RPM tool to install
+the binary RPM to a system with the correct permissions and track which package
+"owns" which files.
+
+DRPM
+^^^^
+
+A binary delta RPM has a payload which contains the binary diff between two
+releases of the same package. For example, there could be a binary delta RPM
+that can be used to upgrade an existing Firefox 45 installation to a Firefox
+46 installation without downloading the entire binary RPM for Firefox 46. This
+format exists to save bandwidth for content provides and requires a significant
+amount of computation on the client attempting to install the DRPM. Unlike RPMs
+and SRPMs, DRPMs are not created by ``rpmbuild``. Other tools exist to build
+them, like `deltarpm <https://github.com/rpm-software-management/deltarpm>`_,
+or `createrepo_c <https://github.com/rpm-software-management/createrepo_c>`_
+which leverages the  `drpm library <https://git.fedorahosted.org/git/drpm.git>`_.
+When using createrepo_c, it is possible to generate DRPMs and the prestodelta.xml
+metadata required for an RPM repository (covered below).
+
+What each RPM contains is completely up to the creator of the RPM package. It
```

Comment:
Perhaps this part should go in the general area, it looks like this is only applicable to drpm.